src/Bicep.Core/IsInternalInit.cs

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace System.Runtime.CompilerServices
+{
+    // This is required to use record types since the project targets 'netstandard 2.1'.
+    // See https://developercommunity.visualstudio.com/t/error-cs0518-predefined-type-systemruntimecompiler/1244809.
```

Comment:
Are there any issues with doing this? It feels a little hacky 😄

Reply:
The answer comes from the .NET team so I feel like it's fine. According to the issue Roslyn will always try to find `IsExternalInit` from source code first: https://github.com/dotnet/roslyn/issues/50696#issuecomment-782188548.

Reply:
Let's add a note to the .net 5 retargeting issue so we don't forget to remove this then.

Reply:
Added a note.